File: maxhanna.client/src/services/datacontracts/ender/meta-hero.ts
import { User } from "../user/user"; 
import { Vector2 } from "./vector2";
import { MetaBot } from "./meta-bot";

export class MetaHero {
  id: number;  
  name?: string;
  position: Vector2; 
  speed: number;
  level: number;
  kills: number;
  map: string;
  color?: string; 
  mask?: number;
  created: Date;

  // created may be a Date or an ISO string from the backend
  constructor(id: number, name: string, position: Vector2, speed: number, map: string, color?: string, mask?: number, level: number = 1, kills: number = 0, created?: Date) {
    this.id = id; 
    this.name = name;
    this.position = position; 
    this.speed = speed;
    this.level = level;
    this.kills = kills;
    this.map = map;
    this.color = color; 
    this.mask = mask;
    this.created = created || new Date();
  } 
}

File: maxhanna.Server/Controllers/DataContracts/Ender/MetaHero.cs
namespace maxhanna.Server.Controllers.DataContracts.Ender
{
    public class MetaHero
    {
        public string? Name { get; set; }
        public int Id { get; set; }
        public int UserId { get; set; }
        public Vector2 Position { get; set; } = new Vector2(0, 0);
        public int Speed { get; set; }
        public string Color { get; set; } = "";
        public int? Mask { get; set; } = null;
        public int Level { get; set; } = 1;
        public int Kills { get; set; } = 0;
        // Server-calculated seconds since hero creation (used by client to set/reset in-game timer)
        public int TimeOnLevelSeconds { get; set; } = 0;
        // Creation timestamp from DB
        public DateTime? Created { get; set; }
    }
}
